### ISSUE TITLE
Receiving duplicate logs

### ISSUE THREAD

**Issue by thair116** [kumavis/eth-json-rpc-filters#5]
Using metamask as my web3 provider, I've set up a listener to watch for emitted events, but notice that each event is being received twice:

<img width="1348" alt="screen shot 2018-03-27 at 1 19 34 pm" src="https://user-images.githubusercontent.com/3717516/37993612-ffc42d84-31c3-11e8-9983-a965fedaaad0.png">

Here is the code that is setting up the listener and logging the above messages: 

```
async function setUpListeners() {
    const tokenInstance = await getContract(token);
    let event = tokenInstance.LinkPosted();
    event.watch((error, result) => {console.log(result)});
}
```

I've confirmed that the contract is only posting one event and that only one event listener is set up.

It looks like the issue is that eth-json-rpc-filters doesn't call `eth_getFilterChanges`, but instead calls `eth_getLogs` and might have an off-by-one error when passing the block number params. Ever time the event listener requests new logs, there is a 1 block overlap between the requested blocks and teh previous blocks:

<img width="896" alt="screen shot 2018-03-27 at 1 20 02 pm" src="https://user-images.githubusercontent.com/3717516/37993852-ad3256c6-31c4-11e8-90e3-d9438a54ad4e.png">

<img width="899" alt="screen shot 2018-03-27 at 1 20 25 pm" src="https://user-images.githubusercontent.com/3717516/37993856-b1aa8aa2-31c4-11e8-97b7-12311fa557c8.png">

It looks like the issue is in this [line](https://github.com/kumavis/eth-json-rpc-filters/blob/master/log-filter.js#L40), where `fromBlock` equals `oldBlock`, instead of `oldBlock+1`.

I'm using metamask 4.3.0 and ganache-cli v6.1.0 (ganache-core: 2.1.0). I do not get this issue when connecting to geth directly as my web3 provider.  I'm also surprised I haven't seen any chatter about this issue, so perhaps I'm just doing something weird...

**Comment by kumavis** [comment]
writing some tests to confirm

**Comment by kumavis** [comment]
hey sorry i was off for a couple weeks, back on this

was able to reproduce the error in tests!
beefed up test suite a bit, now testing against `ganache-core`

**Comment by kumavis** [comment]
fixed in `eth-json-rpc-filters@1.2.6`
